stress/shell/src/main/java/alluxio/stress/cli/fuse/FuseIOBench.java

```diff
@@ -0,0 +1,376 @@
+    for (int i = 0; i < responseTime99Percentile.length; i++) {
+      responseTime99Percentile[i] = (float) methodStatistics.getTimeNs()
+          .getValueAtPercentile(100.0 - 1.0 / (Math.pow(10.0, i))) / Constants.MS_NANO;
+    }
+
+    float[] maxResponseTimesMs = new float[StressConstants.MAX_TIME_COUNT];
+    Arrays.fill(maxResponseTimesMs, -1);
+    for (int i = 0; i < methodStatistics.getMaxTimeNs().length; i++) {
+      maxResponseTimesMs[i] = (float) methodStatistics.getMaxTimeNs()[i] / Constants.MS_NANO;
+    }
+
+    return new SummaryStatistics(methodStatistics.getNumSuccess(),
+        responseTimePercentile,
+        responseTime99Percentile, maxResponseTimesMs);
+  }
+
+  private final class BenchContext {
+    private final long mStartMs;
+    private final long mEndMs;
+
+    /** The results. Access must be synchronized for thread safety. */
+    private FuseIOTaskResult.ThreadCountResult mThreadCountResult;
+
+    public BenchContext(long startMs, long endMs) {
+      mStartMs = startMs;
+      mEndMs = endMs;
+    }
+
+    public long getStartMs() {
+      return mStartMs;
+    }
+
+    public long getEndMs() {
+      return mEndMs;
+    }
+
+    public synchronized void mergeThreadResult(FuseIOTaskResult.ThreadCountResult threadResult) {
+      if (mThreadCountResult == null) {
+        mThreadCountResult = threadResult;
+      } else {
+        try {
+          mThreadCountResult.merge(threadResult);
+        } catch (Exception e) {
+          mThreadCountResult.addErrorMessage(e.getMessage());
+        }
+      }
+    }
+
+    public synchronized FuseIOTaskResult.ThreadCountResult getResult() {
+      return mThreadCountResult;
+    }
+  }
+
+  private final class BenchThread implements Callable<Void> {
+    private final BenchContext mContext;
+    private final List<String> mFilesPath;
+    private final int mThreadId;
+    private final byte[] mBuffer;
+    private final long mFileSize;
+
+    private FileInputStream mInStream = null;
+    private FileOutputStream mOutStream = null;
+    private long mCurrentOffset;
+
+    private final FuseIOTaskResult.ThreadCountResult mThreadCountResult =
+        new FuseIOTaskResult.ThreadCountResult();
+
+    private BenchThread(BenchContext context, int threadId, int numThreads) {
+      mContext = context;
+      mThreadId = threadId;
+      mFilesPath = new ArrayList<>();
+      for (int i = mThreadId; i < mParameters.mNumFiles; i += numThreads) {
+        mFilesPath.add(mParameters.mLocalPath + "/data-" + i);
+      }
+
+      mBuffer = new byte[(int) FormatUtils.parseSpaceSize(mParameters.mBufferSize)];
+      Arrays.fill(mBuffer, (byte) 'A');
+
+      mFileSize = FormatUtils.parseSpaceSize(mParameters.mFileSize);
+    }
+
+    @Override
+    public Void call() {
+      try {
+        runInternal();
+      } catch (Exception e) {
+        LOG.error(Thread.currentThread().getName() + ": failed", e);
+        mThreadCountResult.addErrorMessage(e.getMessage());
+      }
```

Comment:
add a finally block here to make sure if any exception is thrown, we close the opened fileInStream and FileOutStream